### PR TITLE
Add ResNet preprocessing model

### PR DESCRIPTION
**Pull request by jantonguirao**
# ResNet preprocessing model

## Description
It modifies ResNet page to use ONNX for the data preprocessing steps. This is the result of the work from the ONNX preprocessing working-group (see https://github.com/onnx/working-groups/tree/main/preprocessing)

## Model

|Model        |Download  | Download (with sample test data)|ONNX version|Opset version|Accuracy |
|-------------|:--------------|:--------------|:--------------|:--------------|:--------------|
|ResNet-preproc| [4.0KB](preproc/resnet-preproc-v1-18.onnx)  |  [864 KB](preproc/resnet-preproc-v1-18.tar.gz) |  1.13.1 | 18| N/A|

### Source
Created via ONNX parser. See details here: https://github.com/onnx/working-groups/blob/main/preprocessing/notebooks/resnet_preproc_zoo/Resnet%20preprocessing%20model%20and%20data%20preparation.ipynb

## Inference
See https://github.com/onnx/working-groups/blob/main/preprocessing/notebooks/resnet_preproc_zoo/Resnet%20preprocessing%20model%20and%20data%20preparation.ipynb

### Input
Sequence of image in uint8 [H, W, 3] layout

### Preprocessing
This is a preprocessing model

### Output
Single tensor with float32 [N, 3, 224, 224], where N is the number of elements in the input sequence

### Postprocessing
N/A

### Dataset (Train and validation)
N/A

### Training
N/A

### Validation accuracy
N/A

## Test Data Creation

See https://github.com/onnx/working-groups/blob/main/preprocessing/notebooks/resnet_preproc_zoo/Resnet%20preprocessing%20model%20and%20data%20preparation.ipynb

### References
Link to paper or references.

## Contributors
Joaquin Anton (NVIDIA)

## License
Add license information - on default, Apache 2.0
<hr>

### CODE REVIEW

File: vision/classification/resnet/README.md [hunk 1]
```diff
@@ -68,24 +68,88 @@ All pre-trained models expect input images normalized in the same way, i.e. mini
...
+network_model = onnx.version_converter.convert_version(network_model, 18)
+network_model.ir_version = 8
+checker.check_model(network_model)
+
+model_w_preproc = onnx.compose.merge_models(
+    preprocessing_model, network_model,
+    io_map=[('preproc_data', 'data')]
+)
+checker.check_model(model_w_preproc)
+
```
Comment (jantonguirao) [inline vision/classification/resnet/README.md:148]: TODO (separate step). Once https://github.com/onnx/onnx/pull/4976 is merged, add instructions here too

**Review by jcwchen — COMMENTED** [review]
Do you know why ONNX Runtime has this warning:
```
2023-03-10 09:31:44.845[7](https://github.com/onnx/models/actions/runs/4383130901/jobs/7672997997#step:6:8)475 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '_00000132D8B8C750__00000132D8B8B4C0_axes_input'. It is not used by any node and should be removed from the model.
2023-03-10 09:31:44.8465800 [W:onnxruntime:, graph.cc:3490 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer '_00000132D8B8C750__00000132D8B8B4C0_k2'. It is not used by any node and should be removed from the model.
```
although I cannot find such an initializer in the model? Are they created somehow by the local function from ONNX Runtime?

**Comment by jantonguirao** [comment]
@jcwchen Seems that those come from the expansion of CenterCropPad:
https://github.com/onnx/onnx/blob/90eea9c0d0d577169d4055b81f85d0ba59f6baff/onnx/defs/tensor/defs.cc#L3550-L3706

However, looking at the code, it seems that those are actually used, so I am not quite sure what's going on there. Can you perhaps look at the function definition above and let me know if you thing this usage of constants is reasonable?

**Comment by jcwchen** [comment]
Thank you for pointing out! I didn't realized that these initializers are actually created by onnx.shape_inference on the fly. Possibly it's a false alarm (warning) from ONNX Runtime that these initializers are used in fact.

File: workflow_scripts/onnx_test_data_utils.py [hunk 2]
```diff
@@ -11,15 +11,23 @@
...
     np_array = numpy_helper.to_array(tensor)
     return tensor.name, np_array
 
+def read_sequenceproto_pb_file(filename):
+    """Return tuple of sequence name and list of numpy.ndarray of the data from a pb file containing a SequenceProto."""
+    seq = SequenceProto()
+    with open(filename, 'rb') as f:
+        seq.ParseFromString(f.read())
+    list_of_arrays = numpy_helper.to_list(seq)
+    return seq.name, list_of_arrays
```
Comment (jcwchen) [inline workflow_scripts/onnx_test_data_utils.py:29]: Thank you for extending onnx_test_data_utils.py!

Not relevant to this PR: it seems worthy to have the same extension in the original code as well: https://github.com/microsoft/onnxruntime/blob/main/tools/python/onnx_test_data_utils.py.

**Review by jcwchen — APPROVED** [review]

**Comment by jantonguirao** [comment]
> @jantonguirao Thank you very much for contributing to ONNX model-zoo from last year!
> 
> We are updating tutorial/model [contributor list](https://github.com/onnx/sigs/blob/main/CONTRIBUTORS) with contributors from last year. I wonder if there is anyone that has made significant contribution to models in your PR, please let me know so I can add them to the list.
> 
> Thank you!
> 
> Liqun

Just me for this one. Thank you.

**Pull request merged**